cpp/include/cudf/ast/linearizer.hpp

```diff
@@ -0,0 +1,436 @@
+/*
```

Comment:
Same here. Too much `detail` stuff is in this header. Makes it difficult for a public consumer to see what interfaces they need to interact with.

Reply:
I have remedied this to a large extent and will tag you in some follow-up comments where I would like further input.